vulnfeeds/cmd/combine-to-osv/main.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"bufio"
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"github.com/google/osv/vulnfeeds/vulns"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/osv/vulnfeeds/cves"
+)
+
+const (
+	cveUrlBase     = "https://nvd.nist.gov/feeds/json/cve/1.1/"
+	fileNameBase   = "nvdcve-1.1-"
+	startingYear   = 2002
+	cveOutputPath  = "output2"
+	partsInputPath = "parts"
+	osvOutputPath  = "osv_output"
+)
+
+func main() {
+	//jsonPath := flag.String("nvd_json", "", "Path to NVD CVE JSON.")
+	flag.Parse()
+
+	currentYear := time.Now().Year()
+	err := os.MkdirAll(cveOutputPath, 0755)
+	if err != nil {
+		log.Fatalf("Can't create output path: %s", err)
+	}
+	err = os.MkdirAll(osvOutputPath, 0755)
+	if err != nil {
+		log.Fatalf("Can't create output path: %s", err)
+	}
+
+	for i := startingYear; i <= currentYear; i++ {
+		downloadCveAsNeeded(strconv.Itoa(i))
+	}
+	downloadCveAsNeeded("modified")
```

Comment:
In general it's better to build more modular programs here, i.e. breaking up the downloading into a separate thing and passing that as a flag. To start with this could even be in bash right? 

This makes this easier to test (i.e. we can just provide a file) and extensible in the future.

Reply:
That makes sense, I'll pull it out into a separate program